lib/Conversion/TritonToLinearAlgebraSubprograms/TritonToLinearAlgebraSubprogramsPass.cpp

```diff
@@ -0,0 +1,173 @@
+    Value memrefConst = rewriter.create<bufferization::ToMemrefOp>(loc, MemRefType::get(tensorC.getShape(), tensorC.getElementType()), C);
+    auto memrefType = MemRefType::get(tensorC.getShape(), floatType);
+    Value memrefC = rewriter.create<memref::AllocOp>(loc, memrefType);
+    auto copyOp = rewriter.create<linalg::CopyOp>(loc, ValueRange{memrefConst}, ValueRange{memrefC});
+
+    ModuleOp module = op->getParentOfType<ModuleOp>();
+
+    auto intType = rewriter.getI32Type();
+    auto int64Type = rewriter.getI64Type();
+    auto ptrType = LLVM::LLVMPointerType::get(op.getContext(), 0); // default address space
+
+    auto funcType = FunctionType::get(op.getContext(),
+        {intType, intType, intType, intType, intType, intType, floatType,
+         ptrType, intType, ptrType, intType, floatType,
+         ptrType, intType}, {});
+
+    bool usingF64 = floatType.isF64();
+    const char *funcName = usingF64 ? "cblas_dgemm" : "cblas_sgemm";
+    auto func = module.lookupSymbol<func::FuncOp>(funcName);
+    if (!func) {
+      OpBuilder::InsertionGuard guard(rewriter);
+      rewriter.setInsertionPointToStart(module.getBody());
+      func = rewriter.create<func::FuncOp>(loc, funcName, funcType);
+      func.setVisibility(SymbolTable::Visibility::Private);
+    }
+
+    auto memrefToPointer = [&rewriter, &loc, &int64Type, &ptrType](Value &memref) {
+      auto indexPtr = rewriter.create<memref::ExtractAlignedPointerAsIndexOp>(loc, memref);
+      auto castOp = rewriter.create<arith::IndexCastOp>(loc, int64Type, indexPtr);
+      return rewriter.create<LLVM::IntToPtrOp>(loc, ptrType, castOp);
+    };
+
+    auto tensorToPointer = [&rewriter, &loc, &memrefToPointer](Value &V, RankedTensorType &T) {
+      Value memref = rewriter.create<bufferization::ToMemrefOp>(loc, MemRefType::get(T.getShape(), T.getElementType()), V);
+      return memrefToPointer(memref);
+    };
+
+    Value ptrA = tensorToPointer(A, tensorA);
+    Value ptrB = tensorToPointer(B, tensorB);
+    Value ptrC = memrefToPointer(memrefC);
+
+    int32_t M = tensorA.getShape()[0];
+    int32_t K = tensorA.getShape()[1];
+    int32_t N = tensorB.getShape()[1];
+
+    Value alpha = rewriter.create<arith::ConstantOp>(loc, floatType, usingF64 ? rewriter.getF64FloatAttr(1.0) : rewriter.getF32FloatAttr(1.0));
+    Value beta = alpha;
+
+    auto constOp = [&rewriter, &loc, &intType](int32_t V) {
+      return rewriter.create<arith::ConstantOp>(loc, intType, rewriter.getI32IntegerAttr(V));
+      };
+    Value CblasRowMajor = constOp(101), CblasNoTrans = constOp(111);
+    Value MVal = constOp(M), NVal = constOp(N), KVal = constOp(K);
+    Value LDA = KVal, LDB = NVal, LDC = NVal;
+
+    auto funcOp = rewriter.create<func::CallOp>(loc, func, ValueRange{
+        CblasRowMajor, CblasNoTrans, CblasNoTrans,
+        MVal, NVal, KVal,
+        alpha, ptrA, LDA,
+        ptrB, LDB, beta,
+        ptrC, LDC
+    });
+
```

Comment:
i think it would be useful to document the functioin signature here for future reference. e.g.: a link to https://github.com/OpenMathLib/OpenBLAS/blob/3ee9e9d8d050dfe6bb2733a8a87540eb7f70ff56/cblas.h#L62 would help understand where the 101 and 111 come from.

Reply:
done